FEATURE: add option to exclude child document nodes

batabana opened the pull request:
We had a use case in a project where we needed to export only a specific page itself without any child pages. That's why we added the option to do exactly this (`--exclude-child-documents`).
On the way, some deprecated classes were replaced and the project was declared compatible with the current flow version.

Kleisli commented:
Hi @kdambekalns, I just realised, this pull request is open since July.
Is this package still maintained by flownative?

kdambekalns commented:
Thanks for the reminder… it is maintained on an as-needed basis, since we don't use it ourselves. I had overlooked this one…

Kleisli commented:
I might add some more pull requests soon, as I'm currently using it in a larger project. Just have to figure out, what the translation agency needs to work optimally.

kdambekalns commented:
I'll look at this when #17 is merged, to have the Neos 7 changes "out of the way".

batabana committed:
FEATURE: add option to exclude child document nodes

batabana committed:
TASK: Declare compatible with Neos 8

batabana commented:
I dropped the Neos 7 changes as they were handled by someone else in the meantime. Also updated the feature code to work with the refactored ExportService.

kdambekalns commented:
Oh, how time flies… sorry!

kdambekalns committed:
Streamline code a little

This makes the code to collect content for a document a little less
generic, but more concise.

kdambekalns commented:
Released with v2.2.0
